### DISCUSSION TITLE
emqx broker mem usage problem

### DISCUSSION THREAD

**Discussion by wttt-py** [emqx/emqx#7288] [Q&A]
The emqx broker has a strange phenomenon during the stress test. It occupies memory (M) and the CPU usage（%） is always normal. What happened to these troughs is shown in the chart below. There is no problem with log data. Can anyone explain, thanks.
![1646982984(1)](https://user-images.githubusercontent.com/83765757/157821468-bd1436ce-e6cb-456c-b123-e8e30931a67d.jpg)
![1646983320(1)](https://user-images.githubusercontent.com/83765757/157821471-1f65f395-19d1-4043-9311-3e71d8d92e1e.jpg)
![1646983428(1)](https://user-images.githubusercontent.com/83765757/157821477-645a8d0a-fd63-4898-b211-4113c1c86d5b.jpg)

**Comment by lafirest** [comment]
I think it is the GC freed some unused memory

**Reply by wttt-py** [reply]
He always appears in an instant....

**Reply by ieQu1** [reply]
EMQX has periodic GC feature, its behavior matches the pattern.

**Reply by wttt-py** [reply]
thanks for answer
